### PR TITLE
Output last transition author as case worker

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Display author name of last transition for journey in case worker column

- [X] allocated claims display case worker to whom claim is allocated
- [X] submitted, redetermination or awaiting written reasons claim states display 'n/a' for case worker
- [X] refused, rejected, part_authorised and authorised display the case worker that transitioned to that state

#### Ticket

[CBO-1595](https://dsdmoj.atlassian.net/browse/CBO-1595)

#### Why
Case workers are requesting that they see
the case worker who made the assessment
for that "journey". Current case worker
output for a case worker is the "author"
of last decision on the CLAIM, rather
than the journey.

### CODE REVIEW

File: app/services/stats/management_information_generator.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@ def headers
 
     def active_non_draft_claims
       Claim::BaseClaim.active.non_draft
+      # Claim::BaseClaim.where('original_submission_date BETWEEN ? AND ?', Date.parse('20/04/2020').beginning_of_day,Date.parse('20/04/2020').end_of_day)
```
Comment (jsugarman) [inline app/services/stats/management_information_generator.rb]: remove

File: spec/presenters/management_information_presenter_spec.rb [hunk 2]
```diff
@@ -1,455 +1,571 @@
...
-            transition = claim.last_decision_transition
-            transition.update_author_id(another_user.id)
-            claim.redetermine!
-            presenter = ManagementInformationPresenter.new(claim, view)
-            presenter.present! do |claim_journeys|
-              expect(presenter.af1_lf1_processed_by).to eq another_user.name
-            end
+          travel_to(3.days.ago) do
+            set_amount_assessed(claim)
+            claim.authorise_part!(author_id: case_worker2.user.id, reason_code: ['other'], reason_text: 'I am part authorising')
```
Comment (jsugarman) [inline spec/presenters/management_information_presenter_spec.rb]: remove reason as this is not required/expected

File: spec/presenters/management_information_presenter_spec.rb [hunk 3]
```diff
@@ -1,455 +1,571 @@
...
+          end
+
+          travel_to(2.days.ago) { claim.redetermine! }
+
+          travel_to(1.day.ago) do
+            claim_allocator_call!(allocator_options.merge(case_worker_id: case_worker3.id))
+          end
+
+          set_amount_assessed(claim)
+          claim.authorise!(author_id: case_worker3.user.id, reason_code: ['other'], reason_text: 'I am authorising')
```
Comment (jsugarman) [inline spec/presenters/management_information_presenter_spec.rb]: remove reason as this is not required/expected

**Comment by jrmhaig** [comment]
👍 Looks OK based on my (currently) limited (but growing) knowledge of how claim transitions work.

**Comment by kmahern** [comment]
lgtm 👍

**Review by kmahern — APPROVED** [review]
lgtm 👍

**Pull request merged**